test/perf/perf_s3_client.cc

```diff
@@ -0,0 +1,149 @@
+    {}
+
+    future<> start() {
+        plog.debug("Creating {} of {} bytes", _object_name, _object_size);
+
+        auto out = output_stream<char>(_client->make_upload_sink(_object_name));
+        std::exception_ptr ex;
+        try {
+            auto rnd = tests::random::get_bytes(chunk_size);
+            uint64_t written = 0;
+            do {
+                co_await out.write(reinterpret_cast<char*>(rnd.begin()), std::min(_object_size - written, rnd.size()));
+                written += rnd.size();
+            } while (written < _object_size);
+            co_await out.flush();
+        } catch (...) {
+            ex = std::current_exception();
+            plog.error("Cannot write object: {}", ex);
+        }
+        co_await out.close();
+        if (ex) {
+            co_await coroutine::return_exception_ptr(std::move(ex));
+        }
+    }
+
+private:
+
+    future<> do_run() {
+        auto until = now() + _duration;
+        uint64_t off = 0;
+        do {
+            auto start = now();
+            try {
+                co_await _client->get_object_contiguous(_object_name, s3::range{off, chunk_size});
+                off = (off + chunk_size) % (_object_size - chunk_size);
+                _reads_hist.add(std::chrono::duration_cast<std::chrono::milliseconds>(now() - start).count());
+            } catch (...) {
+                _errors++;
+            }
+        } while (now() < until);
+    }
+
+public:
+    future<> run() {
+        co_await coroutine::parallel_for_each(boost::irange(0u, _parallel), [this] (auto fnr) -> future<> {
+            plog.debug("Running {} fiber", fnr);
+            co_await seastar::sleep(std::chrono::milliseconds(fnr)); // make some discrepancy
+            co_await do_run();
+        });
+    }
+
+    future<> stop() {
+        plog.debug("Removing {}", _object_name);
+        co_await _client->delete_object(_object_name);
+        co_await _client->close();
+
+        auto print_percentiles = [] (const utils::estimated_histogram& hist) {
+            return format("min: {:-6d}, 50%: {:-6d}, 90%: {:-6d}, 99%: {:-6d}, 99.9%: {:-6d}, max: {:-6d} [ms]",
+                hist.percentile(0),
+                hist.percentile(0.5),
+                hist.percentile(0.9),
+                hist.percentile(0.99),
+                hist.percentile(0.999),
+                hist.percentile(1.0)
+            );
+        };
+        plog.info("reads total: {:5}, errors: {:5}; latencies: {}", _reads_hist._count, _errors, print_percentiles(_reads_hist));
+    }
+};
+
+int main(int argc, char** argv) {
+    namespace bpo = boost::program_options;
+    app_template app;
+    app.add_options()
+        ("duration", bpo::value<unsigned>()->default_value(10), "seconds to run")
+        ("parallel", bpo::value<unsigned>()->default_value(1), "number of parallel fibers")
+        ("object_size", bpo::value<size_t>()->default_value(1 << 20), "size of test object")
+    ;
```

Comment:
worth allowing the user to configure chunk size, as sstable compression data source uses the configured chunk size, which defaults to 64k IIRC.